src/vs/platform/terminal/node/terminalEnvironment.ts

```diff
@@ -87,10 +87,15 @@ export function getShellIntegrationInjection(
 	if (options.shellIntegration.nonce) {
 		envMixin['VSCODE_NONCE'] = options.shellIntegration.nonce;
 	}
-	if ((!isWindows && shellLaunchConfig.shellIntegrationEnvironmentReporting) || (isWindows && (
-		options.windowsUseConptyDll || options.windowsEnableConpty && getWindowsBuildNumber() >= 22631
-	))) {
-		envMixin['VSCODE_SHELL_ENV_REPORTING'] = '1';
+
+	const enableWindowsEnvReporting = options.windowsUseConptyDll || options.windowsEnableConpty && getWindowsBuildNumber() >= 22631;
+
+	if (shellLaunchConfig.shellIntegrationEnvironmentReporting) {
+		if (isWindows && enableWindowsEnvReporting) {
+			envMixin['VSCODE_SHELL_ENV_REPORTING'] = '1';
+		} else {
+			envMixin['VSCODE_SHELL_ENV_REPORTING'] = '1';
+		}
```

Comment:
This would set it to `1` when `isWindows && !enableWindowsEnvReporting`. Also pull the variable above so it's only run when it's used. Something like this:

```ts
if (isWindows) {
  const enableWindowsEnvReporting = options.windowsUseConptyDll || options.windowsEnableConpty && getWindowsBuildNumber() >= 22631;
  ...
}
```

Reply:
Ahhh, just saw this, let me make it better also looking at the comment below